.github/workflows/ci.yml

```diff
@@ -596,91 +596,13 @@ jobs:
       - name: Fail early
         if: failure()
         uses: andymckay/cancel-action@0.3
-    
-  update-client-api:
-    runs-on: ubuntu-latest
-    needs:
-      - parachain-build-dev
-      - tee-build
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Pull polkadot image
-        run: |
-          docker pull parity/polkadot
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: litentry-parachain-dev
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: litentry-tee
-
-      - name: Load docker image
-        run: |
-          docker load < litentry-parachain-dev.tar.gz
-          docker load < litentry-tee.tar.gz
-          docker images
-
-      - name: Enable corepack and pnpm
-        run: corepack enable && corepack enable pnpm
-        
-      - name: Generate parachain artefacts
-        run: |
-          ./tee-worker/scripts/litentry/generate_parachain_artefacts.sh
-
-      - name: Build litentry parachain docker images
-        run: |
-          cd tee-worker/docker
-          docker compose -f litentry-parachain.build.yml build
-        
-      - name: Update metadata and generate types
-        timeout-minutes: 10
-        run: |
-          cd tee-worker/docker
-          docker compose -f docker-compose.yml -f lit-ts-api-package-build.yml up --no-build --exit-code-from lit-ts-api-package-build lit-ts-api-package-build
-      
-      - name: Zip client-api
-        working-directory: ./tee-worker
-        run: |
-         sudo rm -rf client-api/node_modules client-api/parachain-api/node_modules client-api/sidechain-api/node_modules
-         tar -zcvf client-api.tar.gz client-api
 
-      - name: Upload client-api as artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: client-api
-          path: tee-worker/client-api.tar.gz
-          if-no-files-found: error
-          
-      - name: Stop docker containers
-        run: |
-          cd tee-worker/docker
-          docker compose -f docker-compose.yml -f lit-ts-api-package-build.yml stop
-
-      - name: Collect docker logs if test fails
-        continue-on-error: true
-        uses: jwalton/gh-docker-logs@v2
-        if: failure()
-        with:
-          tail: all
-          dest: docker-logs
-
-      - name: Upload docker logs if test fails
-        uses: actions/upload-artifact@v3
-        if: failure()
-        with:
-          name: build-client-api-docker-logs
-          path: docker-logs
-          if-no-files-found: ignore
-          retention-days: 3
-      
   tee-test:
     runs-on: ubuntu-latest
+    if: needs.set-condition.outputs.run_tee_test == 'true'
     needs:
       - set-condition
-      - update-client-api
+      - tee-build
```

Comment:
I don't think we need `tee-build` here. e.g.: only ts-tests changes, so only need to run tee-test.

Reply:
we still need it as in tee-build there is a step to upload new build or docker build and only after this step we should trigger tee-test
this step `.github/workflows/ci.yml:465`